Jenkinsfile

```diff
@@ -14,13 +14,53 @@
  * limitations under the License.
  */
 
+
+//node-affinity
+//nodes 1..3 are reserved for Jenkins slave pods.
+//node 0 is used for the Jenkins master
+//node 4 is used for SQ and Nexus
+//dind-daemon
+//dind-daemon below is the sidecar creation pattern for the docker in docker entity
+//it allows the creation and build of docker images
+//but more importantly allows the use of testcontainers
+//Be careful with upgrading the version number for the dind-daemon
+//in experiments carried out, the security priviledges seemed to prevent
+//connection to the docker sock for much later versions
+
+
 podTemplate(yaml: '''
 apiVersion: v1
 kind: Pod
+metadata: 
+    name: dind 
 spec:
+  affinity:
+    nodeAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 1
+        preference:
+          matchExpressions:
+          - key: palisade-node-name
+            operator: In
+            values: 
+            - node1
+            - node2
+            - node3
   containers:
+  - name: jnlp
+    image: jenkins/jnlp-slave
+    imagePullPolicy: Always
+    args: 
+    - $(JENKINS_SECRET)
+    - $(JENKINS_NAME)
+    resources:
+      requests:
+        ephemeral-storage: "4Gi"
+      limits:
+        ephemeral-storage: "8Gi"
+
   - name: docker-cmds
-    image: jnlp-did:jdk11
+    image: 779921734503.dkr.ecr.eu-west-1.amazonaws.com/jnlp-did:INFRA
```

Comment:
Does this need to specify the ECR? If so can we parameterise this?

Reply:
Since we are using a real Kubernetes installation rather than microk8s, it is necessary to specify the full tag for the ECR.
I have tried parameterising the variable using ${}, ${env.}, $(), $(env.) - none of which result in the substitution working at all.
The intention, is to incorporate the pod template definition into the Jenkins template longer term so that the Jenkinsfiles are sparse, so the visibility of the ECR registry should not remain an issue.
I can spend more time looking at this but it is not as obvious as it would first appear.